packages/server/lib/browsers/cri-client.ts

```diff
@@ -252,6 +254,31 @@ export const create = async (
       debug('crash detected')
       crashed = true
     })
+
+    // We only want to try and add service worker traffic if we have a host set. This indicates that this is the child cri client.
+    if (host) {
+      await cri.send('Target.setDiscoverTargets', { discover: true })
```

Comment:
Should this be after the `targetCreated` listener? Also, do we really need to `await` this call?

Reply:
Yes good catch (though I don't think it ultimately matters since we won't get new service workers til later). Will update. We do want to await it as we don't want to progress to start the runs until we're done setting this.

Reply:
[`987f3d3` (#28060)](https://github.com/cypress-io/cypress/pull/28060/commits/987f3d32c7cc87e0163f174f6ba458b577eac597)